lib/mongo/protocol/update.rb

```diff
@@ -58,7 +58,7 @@ def initialize(database, collection, selector, update, options = {})
         @selector    = selector
         @update      = update
         @flags       = options[:flags] || []
-        @upconverter = Upconverter.new(collection, selector, update, flags)
+        @upconverter = Upconverter.new(collection, selector, update, flags, options[:array_filters])
```

Comment:
None of the changes in this file are necessary, as array filters will only be used with write commands.